### PR TITLE
feat: add command line interface

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

Adds an `axe-sarif-converter` CLI tool to wrap the library and associated acceptance tests. Proposed usage:

```
> axe-sarif-converter --help
axe-sarif-converter: Converts JSON files containing axe-core Result object(s)
into SARIF files

Options:
  --help             Show help                                         [boolean]
  --version          Show version number                               [boolean]
  --input-files, -i  Input JSON file(s) containing axe-core Result object(s).
                     Does not support globs. Each input file may consist of
                     either a single root-level axe-core Results object or a
                     root-level array of axe-core Results objects.
                                                              [array] [required]
  --output-file, -o  Output SARIF file. Multiple input files (or input files
                     containing multiple Result objects) will be combined into
                     one output file with a SARIF Run per axe-core Result.
                                                             [string] [required]
  --verbose, -v      Enables verbose console output.  [boolean] [default: false]
  --pretty, -p       Includes line breaks and indentation in the output.
                                                      [boolean] [default: false]
  --force, -f        Overwrites the output file if it already exists.
                                                      [boolean] [default: false]

Examples:
  axe-sarif-converter -i axe-results.json -o axe-results.sarif
```

Looking for a review of both the command line interface and also the code.

#### Pull request checklist

- [x] PR title respects [Conventional Commits](https://www.conventionalcommits.org) (starts with `fix:`, `feat:`, etc, and is suitable for user-facing release notes)
- [x] PR contains no breaking changes, **OR** description of both PR **and final merge commit** starts with `BREAKING CHANGE:`
- [n/a] (if applicable) Addresses issue: #0000
- [x] Added relevant unit tests for your changes
- [x] Ran `yarn precheckin`
- [x] Verified code coverage for the changes made

### CODE REVIEW

**Comment by dbjorge** [comment]
The other option for this would be to use a separate `axe-sarif-converter-cli` package for it; this would avoid the yargs dependency for non-CLI users but would be a little more complex to manage. I imagine virtually all users of this package will use it as a devDependency (where the extra transitive dep isn't so important), so my inclination would be to do the simpler option in the same package, but I could be convinced otherwise.

File: README.md [hunk 1]
```diff
@@ -46,6 +46,18 @@ test('my accessibility test', async () => {
...
+```bash
+# axe-cli is used here for example purposes only; you could also run axe-core
+# using your library of choice and JSON.stringify the results.
+npx axe-cli https://accessibilityinsights.io --save ./sample-axe-results.json
+
+npx axe-sarif-converter --input-files ./sample-axe-results.json --output-file ./sample-axe-results.sarif
+```
+
+See `npx axe-sarif-converter --help` for full command line option details.
+
```
Comment (peterdur) [inline README.md]: Is it worth calling out somewhere that the way to run the CLI after building in this repo is to do `node dist/cli.js (args)`? I ended up looking at the test code to figure this out.

Comment (dbjorge) [reply README.md]: Good idea, added

File: src/cli.test.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            await invokeCliWith(`-o irrelevant.sarif`);
+            fail('Should have returned non-zero exit code');
+        } catch (e) {
+            expect(e.stderr).toMatch('Missing required argument: input-files');
+        }
+    });
+
+    it('requires the -o parameter', async () => {
+        const outputFile = path.join(testResultsDir, 'overwrite_test.sarif');
+        await writeFile(outputFile, 'preexisting content');
```
Comment (smoralesd) [inline src/cli.test.ts]: is this step relevant to the test case? I'm assuming `yargs` will check required parameters before actually try to read files, is that correct?

File: src/cli.test.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+    it('prints version number from package.json with --version', async () => {
+        const output = await invokeCliWith('--version');
+        expect(output.stderr).toBe('');
+        expect(output.stdout).toBe('0.0.0-managed-by-semantic-release\n');
+    });
+
+    it('requires the -i parameter', async () => {
+        const outputFile = path.join(testResultsDir, 'overwrite_test.sarif');
+        await writeFile(outputFile, 'preexisting content');
```
Comment (smoralesd) [inline src/cli.test.ts]: is this step relevant to the test case? I'm assuming `yargs` will check required parameters before actually try to read files, is that correct?

Comment (dbjorge) [reply src/cli.test.ts]: Copy-paste error, fixed!

File: src/cli.test.ts (hunk 2 above)
Comment (dbjorge) [reply src/cli.test.ts]: Copy-paste error, fixed!
